app-frontend/src/app/components/diagramContainer/diagramContainer.controller.js

```diff
@@ -343,22 +310,41 @@ export default class DiagramContainerController {
                         }
                     }
                 },
-                items: this.createPorts(config.inputs, config.outputs)
+                items: inputs
             }
         });
 
         this.shapes.push(shape);
-        return id;
+        this.nodes.set(id, config);
+
+        if (config.children) {
+            let inputPorts = inputs.filter(i => i.group === 'inputs');
+            config.children.forEach((c, idx) => {
+                this.createLink([c.id, 'Output'], [shape.id, inputPorts[idx].id]);
+            });
+        }
+        return shape;
     }
 
     createPorts(inputs, outputs) {
         let ports = [];
-        inputs.forEach(i => {
-            ports.push({
-                id: i,
-                group: 'inputs'
+        if (Array.isArray(inputs)) {
+            inputs.forEach((i, idx) => {
+                ports.push({
+                    id: `input-${idx}`,
+                    label: `input-${idx}`,
+                    group: 'inputs'
+                });
             });
-        });
+        } else {
+            Array(inputs).fill().forEach(idx => {
+                ports.push({
+                    id: `input-${idx}`,
+                    label: `input-${idx}`,
+                    group: 'inputs'
+                });
+            });
+        }
```

Comment:
To do this functionally, you can do this:
```
let length = Array.isArray(inputs) ? inputs.length : inputs;
let ports = Array(inputs).fill().map((_, i) => {
    return {
	id: `input-${i}`,
	label: `input-${i},
	group: `inputs`
    }
}
```
I think that what you have now will result in id/label being `input-undefined`